### ISSUE TITLE
backprop fails with Conv2d layers with v0.5.1 (duet)

### ISSUE THREAD

**Issue by dhalf** [OpenMined/PySyft#6784]
## Description
Calls to backward on the loss of a model that uses Conv2d layers fail on the server side although they work with Linear layers.
Note that this failure is only visible on the server logs and is silently ignored by the client.

## How to Reproduce
Client code:
```
import torch
import syft as sy

duet = sy.join_duet(loopback=True)

class Test(sy.Module):
    def __init__(self, torch_ref):
        super().__init__(torch_ref)
        self.fc1 = self.torch_ref.nn.Conv2d(1, 1, 1, bias=False)
    
    def forward(self, x):
        return self.fc1(x)

model = Test(torch)
remote_model = model.send(duet)
params = remote_model.parameters()

opt = duet.torch.optim.Adam(params=params, lr=0.1)

input("[Hit ENTER to proceed with training]")

inputs = duet.store[0]
targets = duet.store[1]

for _ in range(100):
    opt.zero_grad()
    outputs = remote_model(inputs)
    loss = duet.torch.nn.functional.mse_loss(outputs, targets)
    loss.backward()
    opt.step()

print(remote_model.parameters().get(request_block=True, reason="", timeout_secs=5))
```

Server code:
```
import syft
import torch

duet = syft.launch_duet(loopback=True)

X = torch.tensor([[1.0], [2.0], [0.5], [0.2]])
y = torch.tensor([[2.0], [4.0], [1.0], [0.4]])

X_ptr = X.send(duet, searchable=True)
y_ptr = y.send(duet, searchable=True)

duet.requests.add_handler(action="accept")
```
## Expected Behavior
The model with the conv layer should work just as the same model with a Linear layer: `self.torch_ref.nn.Linear(1, 1, bias=False)`
The parameter should converge to 2.0

## System Information
 - OS: Ubuntu
 - OS Version: 22.04
 - Language Version: Python 3.10.4
 - Package Manager Version: pip 22.0.2
 - syft version: 0.5.1
 - torch version: 1.12.0

**Comment by madhavajay** [comment]
0.5 is no longer supported.
